torchtune/datasets/_grammar.py

```diff
@@ -47,12 +84,11 @@ def grammar_dataset(
         >>> Batch size: 8
     """
 
-    return instruct_dataset(
-        tokenizer=tokenizer,
+    message_transform = GrammarMessages(train_on_input=train_on_input)
+    ds = FinetuneDataset(
```

Comment:
If a user would like access to the underlying dataset prior to any transformers, e.g. to filter the dataset before sampling, they would have to just define their own dataset? 
This is probably not super relevant here, but just thinking ahead if we want to eventually provide similar kind of abstractions for other kinds of datasets/use-cases, that don't fall neatly into per-sample custom processing.

Reply:
The raw data should still be accessible in the class attribute self.data. Filtering can also be folded into the data transform. Although I may need to add a bit of logic to support ignoring certain samples.